### DISCUSSION TITLE
Expose handler type?

### DISCUSSION THREAD

**Discussion by AustinScola** [aio-libs/aiohttp#5842] [Q&A] [answered]
I'm writing a middleware coroutine in a project which is type-checked with mypy. I want to annotated the handler parameter similar to the following:

```Python3
@middleware
async def my_middleware(request: Request, handler: Handler) -> Response:
    ...
```

I can't seem to find a handler type alias to use in `aiohttp` though I see a few defined throughout the code (I could be mistaken that none are available to use?):
```
$ ack "Handler.*Callable" aiohttp
web_middlewares.py
44:_Handler = Callable[[Request], Awaitable[StreamResponse]]

web_protocol.py
61:_RequestHandler = Callable[[BaseRequest], Awaitable[StreamResponse]]

web_urldispatcher.py
84:_WebHandler = Callable[[Request], Awaitable[StreamResponse]]
85:_ExpectHandler = Callable[[Request], Awaitable[None]]

web_app.py
51:    _Handler = Callable[[Request], Awaitable[StreamResponse]]
60:    _Handler = Callable

web_routedef.py
55:_SimpleHandler = Callable[[Request], Awaitable[StreamResponse]]
```

Maybe one should be added to `aiohttp.typdefs`?

P.S. Happy to contribute!

**Comment by Dreamsorcerer** [comment, marked as answer]
Duplicate #4686

I think it's probably time for someone to try this out in a PR, then we can discuss it there.

**Reply by AustinScola** [reply]
Made PR #5847
